src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/BucketHelper.scala

```diff
@@ -97,14 +97,13 @@ class BucketHelper(dataprocConfig: DataprocConfig,
     } yield ()
   }
 
-  private def getPetOrDefault(googleProject: GoogleProject, serviceAccountInfo: ServiceAccountInfo): Future[Option[GcsEntity]] = {
-    // notebook SA, else cluster SA, else compute engine default SA
-    val transformed =
-      OptionT.fromOption[Future](serviceAccountInfo.notebookServiceAccount) orElse
-        OptionT.fromOption[Future](serviceAccountInfo.clusterServiceAccount) orElse
-          OptionT(gdDAO.getComputeEngineDefaultServiceAccount(googleProject))
+  private def getBucketSAs(googleProject: GoogleProject, serviceAccountInfo: ServiceAccountInfo): Future[List[GcsEntity]] = {
+    // notebook SA, cluster SA, and compute engine default SA, if they exist
 
-    transformed.map(userEntity).value
+    gdDAO.getComputeEngineDefaultServiceAccount(googleProject) map { defaultSaOpt =>
```

Comment:
I was thinking about this -- technically we only need to return the compute engine default if the `clusterServiceAccount` is empty. So we could avoid a Google API call here. I was wondering if this logic could take that into account, something like:

```
  private def getBucketSAs(googleProject: GoogleProject, serviceAccountInfo: ServiceAccountInfo): Future[List[GcsEntity]] = {
    // cluster SA orElse compute engine default SA
    val clusterOrComputeDefault = OptionT.fromOption(serviceAccountInfo.clusterServiceAccount) orElse OptionT(gdDAO.getComputeEngineDefaultServiceAccount(googleProject))

    // List(cluster or default SA, notebook SA) if they exist
    clusterOrComputeDefault.value.map { clusterOrDefaultSAOpt =>
      List(clusterOrDefaultSAOpt, serviceAccountInfo.notebookServiceAccount).flatten.map(userEntity)
    }
}
```

Reply:
OK I'll do that.